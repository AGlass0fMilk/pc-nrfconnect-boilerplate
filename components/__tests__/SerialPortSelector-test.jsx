import React from 'react';
import renderer from 'react-test-renderer';
import {
    decorateSerialPortSelector,
    mapSerialPortSelectorDispatch,
    mapSerialPortSelectorState,
} from '../SerialPortSelector';

describe('SerialPortSelector', () => {
    it('should implement component decorator', () => {
        expect(decorateSerialPortSelector).toBeInstanceOf(Function);
    });
    it('should implement dispatch mapper', () => {
        expect(mapSerialPortSelectorDispatch).toBeInstanceOf(Function);
    });
    it('should implement state mapper', () => {
        expect(mapSerialPortSelectorState).toBeInstanceOf(Function);
    });
});

describe('decorateSerialPortSelector', () => {
    it('should render component', () => {
        const SerialPortSelector = decorateSerialPortSelector('div');
        expect(renderer.create(
            <SerialPortSelector />,
        )).toMatchSnapshot();
    });
});
