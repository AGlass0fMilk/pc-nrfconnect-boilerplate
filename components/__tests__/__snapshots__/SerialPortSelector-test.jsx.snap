exports[`decorateSerialPortSelector should render component 1`] = `<div />`;
